project-codeflare/codeflare-sdk
# Fixed creating ingresses without admin access

# Issue link


Closes #400 
# What changes have been made


Notable changes:
1. Removed the use of getting the ingress domain from `config.openshift.io.ingresses` and opted for the creation of routes when on OpenShift. 
2.  Fixed the creation of ingresses when setting `mcad=False` in cluster configuration
3. On OpenShift users must pass an ingress_domain when using the ray client. (this is mandatory as we need the domain name when creating the cert command in the yaml see [here](https://github.com/project-codeflare/codeflare-sdk/blob/main/src/codeflare_sdk/utils/generate_yaml.py#L443)
4. We now use the discovery api to discern whether the user is on Kind or OpenShift by searching for the `route.openshift.io` api group. (This should resolve cluster level privileges issue)


# Verification steps



1. Run through a notebook on Kind & OpenShift and verify ingresses/routes are created properly
2. Run through a notebook on Kind with `mcad=False` to verify ingresses are created/deleted
## Checks
- Testing Strategy
